inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/manager/InlongMultiTopicManager.java

```diff
@@ -0,0 +1,289 @@
+    private final Map<String, List<TopicFetcher>> kafkaFetchers = new ConcurrentHashMap<>();
+    private final Map<String, List<TopicFetcher>> tubeFetchers = new ConcurrentHashMap<>();
+    private final Map<String, TopicFetcher> allFetchers = new ConcurrentHashMap<>();
+    private Set<String> allTopics = new HashSet<>();
+    private final PeriodicTask updateMetaDataWorker;
+
+    private boolean stopAssign = false;
+    private int consumerSize;
+
+    public InlongMultiTopicManager(ClientContext context, QueryConsumeConfig queryConsumeConfig) {
+        super(context, queryConsumeConfig);
+        this.consumerSize = context.getConfig().getMaxConsumerSize();
+        updateMetaDataWorker = new UpdateMetaDataThread(context.getConfig().getUpdateMetaDataIntervalSec(),
+                TimeUnit.SECONDS);
+        String threadName = "sortsdk_multi_topic_manager_" + context.getConfig().getSortTaskId()
+                + "_" + StringUtil.formatDate(new Date(), "yyyy-MM-dd HH:mm:ss");
+        updateMetaDataWorker.start(threadName);
+        LOGGER.info("create InlongMultiTopicManager success");
+    }
+
+    @Override
+    public boolean clean() {
+        LOGGER.info("start clean {}", context.getConfig().getSortTaskId());
+        close();
+        offlineAllTopicsAndPartitions();
+        LOGGER.info("end clean {}", context.getConfig().getSortTaskId());
+        return true;
+    }
+
+    @Override
+    public TopicFetcher addTopic(InLongTopic topic) {
+        return null;
+    }
+
+    @Override
+    public TopicFetcher removeTopic(InLongTopic topic, boolean closeFetcher) {
+        return null;
+    }
+
+    @Override
+    public TopicFetcher getFetcher(String fetchKey) {
+        return allFetchers.get(fetchKey);
+    }
+
+    @Override
+    public Collection<TopicFetcher> getAllFetchers() {
+        return allFetchers.values();
+    }
+
+    @Override
+    public Set<String> getManagedInLongTopics() {
+        return allTopics;
+    }
+
+    @Override
+    public void offlineAllTopicsAndPartitions() {
+        String subscribeId = context.getConfig().getSortTaskId();
+        try {
+            LOGGER.info("start offline {}", subscribeId);
+            stopAssign = true;
+            Set<Map.Entry<String, TopicFetcher>> entries = allFetchers.entrySet();
+            for (Map.Entry<String, TopicFetcher> entry : entries) {
+                String fetchKey = entry.getKey();
+                TopicFetcher topicFetcher = entry.getValue();
+                boolean succ = false;
+                if (topicFetcher != null) {
+                    try {
+                        succ = topicFetcher.close();
+                    } catch (Exception e) {
+                        LOGGER.error(e.getMessage(), e);
+                    }
+                }
+                LOGGER.info("close fetcher{} {}", fetchKey, succ);
+            }
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
```

Comment:
Suggest adding more detailed info in the error log.

Reply:
fixed, thx